File: Zhockon/python/audio.py
#
# This file is part of Zhockon Platform project.
# 
# Copyright (C) 2009-2011 William Oliveira de Lagos <william.lagos1@gmail.com>
#
# Zhockon is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Zhockon is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Zhockon.  If not, see <http://www.gnu.org/licenses/>.
#

import os.path, sys, time
import pyopenal as OpenAL

def load_sound():
    try:
        raise NotImplementedError
    except NotImplementedError, e:
        print "Option not already implemented."

class OAL:
    def test_openal():
        OpenAL.init()
        try:
            l = OpenAL.Listener(22050)
            l.position = (0.0, 0.0, 0.0)
            l.at = (0.0, 0.0, 1.0)
            l.up = (0.0, 1.0, 0.0)
            
            b = OpenAL.WaveBuffer(os.path.join(os.path.dirname(sys.argv[0]), "sword.wav"))
            #b = OpenAL.OggVorbisBuffer("/home/jiba/src/slune/sounds/summum_of_the_light.ogg")
            
            s = OpenAL.Source()
            s.position = (0.0, 0.0, 0.0)
            s.velocity = (0.0, 0.0, 0.0)
            s.buffer  = b
            s.looping = 1
            
            s.play()
            
            # Move the source further
            
            z = 0.0
            while 1:
                z += 0.5
                s.position = (0.0, 0.0, z)
                time.sleep(0.1)
                
            print 'ok'
        except RuntimeError, e: print "Unavailable device."

    def test_device():
        try:
            raise NotImplementedError
        except NotImplementedError, e:
            print "Option not already implemented."        
        # device = openal.alcOpenDevice(None)
        # device = openal.alcOpenDevice("'( ( devices '( native waveout esd sdl alsa arts null ) ) )")
        # context = openal.alcCreateContext (device, (openal.ALC_FREQUENCY, 22050, openal.ALC_INVALID))
        # openal.alcMakeContextCurrent (context)
            
        # openal.alListenerfv (openal.AL_POSITION, (0.0, 0.0, 0.0))
        # openal.alListenerfv (openal.AL_ORIENTATION, (0.0, 0.0, 1.0, 0.0, 1.0, 0.0))
        # stereo = openal.alGenBuffers (1)
        
        # file = open('./sword.wav', 'r')
        # data = file.read()
        # openal.alBufferData (stereo, openal.AL_FORMAT_WAVE_EXT, data, 0)
        # moving_source = openal.alGenSources (1)
        
        # openal.alSource3f (moving_source, openal.AL_POSITION, 0.0, 0.0, 20.0)
        # openal.alSourcefv (moving_source, openal.AL_VELOCITY, (0.0, 0.0, 0.0))
        # openal.alSourcei  (moving_source, openal.AL_BUFFER, stereo)
        # openal.alSourcei  (moving_source, openal.AL_LOOPING, openal.AL_FALSE)
        
        # openal.alSourcePlayv (moving_source)
        # openal.alSourcei (moving_source, openal.AL_SOURCE_STATE, openal.AL_INITIAL)
        
        # print "Using OpenAL", openal.alGetString(openal.AL_VERSION), "by", openal.alGetString(openal.AL_VENDOR)
        
        # while(1): pass

File: remote/Zhockon/core/audio.cc
/*
 * This file is part of Zhockon Platform project.
 * 
 * Copyright (C) 2009-2010 William Oliveira de Lagos <williamx64@gmail.com>
 *
 * Zhockon is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Zhockon is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Zhockon.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <zhockon.h>
using namespace Zhockon;

void Source::setLooping()
{
  
}
void Source::setSound(int value)
{
  
}
void Source::setGain(float value)
{
  
}
void Source::setAmbient()
{
  
}
void Source::play()
{
  
}
void Source::stop()
{
  
}


Sample::Sample(string)
{
  
}

int Sample::get()
{
  return 0;
}


Sound::Sound(Source *src)
{
	source = src;
}

int Sound::loadStream(string fileName)
{
	try{
		source->stop();
		Sample *sample = new Sample(fileName);
		source->setSound(sample->get());
	}catch(Error e){
		std::cerr << e.getMessage() << "\n";
		return -1;
	}
	return 0;
}
int Sound::playStream()
{
	try{
		source->play();
		source->stop();
	}catch(Error e){
		std::cerr << e.getMessage() << "\n";
		return -1;
	}
	return 0;
}
void Sound::setEffects()
{
	source->setGain(0.6f); 
	source->setLooping();
	source->setAmbient();
}
Source* Sound::getSource()
{
	return source;
}